pkg/component/kubeapiserver/deployment.go

```diff
@@ -828,6 +836,8 @@ func (k *kubeAPIServer) vpnSeedClientContainer(index int) *corev1.Container {
 			},
 		},
 		SecurityContext: &corev1.SecurityContext{
+			RunAsNonRoot: pointer.Bool(false),
+			RunAsUser:    pointer.Int64(0),
```

Comment:
What happens when a Pod has a securityContext set at Pod level and then some container overwrite it? Does the container securityContext is considered only or does kubernetes internally merges the container securityContext with the pod securityContext (where container securityContext values take precedence)? 
I am asking because if a merge happens under the hood, then the merged securityContext would be
```yaml
securityContext:
  runAsNonRoot: false
  runAsUser: 0
  runAsGroup: 65532
  fsGroup: 65532
```

which I think is not what we want.
The docs write 
>  Security settings that you specify for a Container apply only to the individual Container, and they override settings made at the Pod level when there is overlap.

but I am not 100% sure.

Reply:
It seems that a merge happens under the hood indeed. 

A small isolated setup to prove the merge:
1. Create the following yaml:
```yaml
apiVersion: v1
kind: Pod
metadata:
  name: alpine
  namespace: default
spec:
  securityContext:
    runAsNonRoot: true
    runAsUser: 65532
    runAsGroup: 65532
    fsGroup: 65532
  containers:
  - image: alpine:3.13.2
    command:
      - /bin/sh
      - "-c"
      - "sleep 60m"
    imagePullPolicy: IfNotPresent
    name: alpine
    securityContext:
      runAsNonRoot: false
      runAsUser: 0
  restartPolicy: Always
```

2. Verify the merge

```
% k exec -ti alpine -- sh
/ # id
uid=0(root) gid=65532 groups=1(bin),2(daemon),3(sys),4(adm),6(disk),10(wheel),11(floppy),20(dialout),26(tape),27(video),65532
```

Again, maybe this is not what we want? Maybe it is better to move the Pod level securityContext to the kube-apiserver container?

Reply:
Yes, the Pod's `securityContext` setting will be used by every container and only the specified settings will be overridden by the container's `securityContext`. It is important to note that `fsGroup` can be set only at the Pod level settings so the only question is if we should explicitly set `runAsGroup` at the Container level.

I do not think that it is required since we already set `root` as the user which grants full permissions on all files in the container.